adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -579,7 +579,9 @@ func (h *hnsw) removeTombstonesAndNodes(deleteList helpers.AllowList, breakClean
 
 		h.resetLock.Lock()
 		if !breakCleanUpTombstonedNodes() {
+			h.Lock()
 			h.nodes[id] = nil
+			h.Unlock()
```

Comment:
we probably also need the h.shardedNodeLocks[candidate%NodeLockStripe] around this